pkg/hubble/metrics/api/api.go

```diff
@@ -146,3 +128,101 @@ var registry = NewRegistry(
 func DefaultRegistry() *Registry {
 	return registry
 }
+
+// FlowFilters is a slice of filters to include or exclude flows.
+type FlowFilters []*pb.FlowFilter
+
+type ContextValues []string
+
+// ContextOptions is a structure used for configuring Hubble metrics context options
+type ContextOptionConfig struct {
+	Name   string        `json:"name,omitempty" yaml:"name,omitempty"`
+	Values ContextValues `json:"values,omitempty" yaml:"values,omitempty"`
+}
+
+type ContextOptionConfigs []*ContextOptionConfig
+
+type MetricConfig struct {
+	Name                 string               `json:"name,omitempty" yaml:"name,omitempty"`
+	ContextOptionConfigs ContextOptionConfigs `json:"contextOptions,omitempty" yaml:"contextOptions,omitempty"`
+	IncludeFilters       FlowFilters          `json:"includeFilters,omitempty" yaml:"includeFilters,omitempty"`
+	ExcludeFilters       FlowFilters          `json:"excludeFilters,omitempty" yaml:"excludeFilters,omitempty"`
+}
+
+// DynamicMetricsConfig represents the structure used for configuring
+// Hubble metrics context options
+type DynamicMetricsConfig struct {
```

Comment:
@chancez  Can you help pre-review these structures before i start making more changes.